pkg/assembler/backends/ent/backend/artifact.go

```diff
@@ -108,7 +110,7 @@ func upsertBulkArtifact(ctx context.Context, tx *ent.Tx, artInputs []*model.IDor
 
 		err := tx.Artifact.CreateBulk(creates...).
 			OnConflict(
-				sql.ConflictColumns(artifact.FieldDigest),
```

Comment:
Thanks to @nchelluri for diving into this and figuring out the issue was that the artifact was not being created because of the `algorithm` mismatch.